### PR TITLE
Make HistogramStandardization accept primitive arguments

### PR DESCRIPTION
**Pull request by GFabien**
Make `HistogramStandardization` accept primitive arguments by adding a method to parse `landmarks_dict` which accepts either a dictionary of NumPy arrays, a path to a dictionary and a dictionary of paths to NumPy arrays. See https://github.com/fepegar/torchio/issues/159.

### CODE REVIEW

File: torchio/transforms/preprocessing/intensity/histogram_standardization.py [hunk 1]
```diff
@@ -23,21 +24,32 @@ class HistogramStandardization(NormalizationTransform):
     See example in :py:func:`torchio.transforms.HistogramStandardization.train`.
 
     Args:
-        landmarks_dict: Dictionary in which keys are image names in the sample
-            and values are NumPy arrays defining the landmarks after training
-            with :py:meth:`torchio.transforms.HistogramStandardization.train`.
+        landmarks_dict: Dictionary are path to a dictionary in which keys are
```
Comment (fepegar) [inline torchio/transforms/preprocessing/intensity/histogram_standardization.py:27]: are -> or?

File: torchio/transforms/preprocessing/intensity/histogram_standardization.py [hunk 2]
```diff
@@ -23,21 +24,32 @@ class HistogramStandardization(NormalizationTransform):
...
+            image names in the sample and values are NumPy arrays or paths to
+            NumPy arrays defining the landmarks after training with
```
Comment (fepegar) [inline torchio/transforms/preprocessing/intensity/histogram_standardization.py:29]: Could you please add some examples in the docstring?

File: torchio/transforms/preprocessing/intensity/histogram_standardization.py [hunk 3]
```diff
@@ -23,21 +24,32 @@ class HistogramStandardization(NormalizationTransform):
...
             masking_method: Union[str, TypeCallable, None] = None,
             p: float = 1,
             ):
         super().__init__(masking_method=masking_method, p=p)
-        self.landmarks_dict = landmarks_dict
+        self.landmarks_dict = self.parse_landmarks_dict(landmarks_dict)
+
+    @staticmethod
+    def parse_landmarks_dict(landmarks_dict):
+        if isinstance(landmarks_dict, str):
```
Comment (fepegar) [inline torchio/transforms/preprocessing/intensity/histogram_standardization.py:46]: What if it's an instance of `Path`?

File: torchio/transforms/preprocessing/intensity/histogram_standardization.py [hunk 4]
```diff
@@ -23,21 +24,32 @@ class HistogramStandardization(NormalizationTransform):
...
+            with open(landmarks_dict, 'rb') as file:
```
Comment (fepegar) [inline torchio/transforms/preprocessing/intensity/histogram_standardization.py:47]: Could you please just do `torch.load(landmarks_dict)`?

File: torchio/transforms/preprocessing/intensity/histogram_standardization.py [hunk 5]
```diff
@@ -23,21 +24,32 @@ class HistogramStandardization(NormalizationTransform):
...
+                landmarks_dict = pickle.load(file)
+        for key, value in landmarks_dict.items():
+            if isinstance(value, str):
```
Comment (fepegar) [inline torchio/transforms/preprocessing/intensity/histogram_standardization.py:50]: Same as above, what if `Path`?

File: torchio/transforms/preprocessing/intensity/histogram_standardization.py (hunk 1 above)
Comment (GFabien) [reply torchio/transforms/preprocessing/intensity/histogram_standardization.py:27]: Oups! You're right!

**Comment by GFabien** [comment]
I agree with all your remarks, I will do that!

**Comment by fepegar** [comment]
I added some minor changes regarding type hints, some a bit off-topic.

**Comment by fepegar** [comment]
Thanks again, @GFabien!

**Pull request merged**